GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1293,61 +1277,43 @@ void DoCommit()
-                        Icon = Microsoft.WindowsAPICodePack.Dialogs.TaskDialogStandardIcon.Error,
-                        Cancelable = true,
+                        Buttons = { TaskDialogButton.Cancel },
+                        Icon = TaskDialogIcon.Error,
+                        AllowCancel = true,
+                        SizeToContent = true
                     };
-                    TaskDialogCommandLink btnCheckout = new("Checkout", null, TranslatedStrings.ButtonCheckoutBranch);
-                    btnCheckout.Click += (s, e) =>
-                    {
-                        dialogResult = 0;
-                        dialog.Close();
-                    };
-                    TaskDialogCommandLink btnCreate = new("Create", null, TranslatedStrings.ButtonCreateBranch);
-                    btnCreate.Click += (s, e) =>
-                    {
-                        dialogResult = 1;
-                        dialog.Close();
-                    };
-                    TaskDialogCommandLink btnContinue = new("Continue", null, TranslatedStrings.ButtonContinue);
-                    btnContinue.Click += (s, e) =>
-                    {
-                        dialogResult = 2;
-                        dialog.Close();
-                    };
-                    dialog.Controls.Add(btnCheckout);
-                    dialog.Controls.Add(btnCreate);
-                    dialog.Controls.Add(btnContinue);
+                    TaskDialogCommandLinkButton btnCheckout = new(TranslatedStrings.ButtonCheckoutBranch);
+                    TaskDialogCommandLinkButton btnCreate = new(TranslatedStrings.ButtonCreateBranch);
+                    TaskDialogCommandLinkButton btnContinue = new(TranslatedStrings.ButtonContinue);
+                    page.Buttons.Add(btnCheckout);
+                    page.Buttons.Add(btnCreate);
+                    page.Buttons.Add(btnContinue);
 
-                    dialog.Show();
-
-                    switch (dialogResult)
+                    TaskDialogButton result = TaskDialog.ShowDialog(Handle, page);
+                    if (result == TaskDialogButton.Cancel)
                     {
-                        case 0:
-                            var revisions = _editedCommit is not null ? new[] { _editedCommit.ObjectId } : null;
-                            if (!UICommands.StartCheckoutBranch(this, revisions))
-                            {
-                                return;
-                            }
-
-                            break;
-                        case 1:
-                            if (!UICommands.StartCreateBranchDialog(this, _editedCommit?.ObjectId))
-                            {
-                                return;
-                            }
+                        return;
+                    }
 
-                            break;
-                        case -1:
+                    if (result == btnCheckout)
```

Comment:
I prefer a case with this many options.
In a few other places too

Reply:
Do you want to use separate variable for dialog result, like `int dialog result = -1` or maybe to use `Tag` property of each button? I thought that it would be good to use the same approach in all places. For me switch/case with values like `1` , `2` is not very intuitive. On the other hand the `if` statements with checks like `result == btnCheckout` is more self explanatory.
But I don't mind to make changes. What option is more preferable - `int` variable or `Tag` ?

Reply:
No change requested. 
It is better to avoid the additional tags etc. In the back of my mind I was thinking about C#8 switch expressions, but that is still requiring constants...
Good work.